### PR TITLE
More listops tests

### PR DESCRIPTION
**Pull request by pmario**
Working on the new "story related" functions I needed several new lisops test, to be sure not breaking stuff. 

Some of the tests do have weird results. ... See comments. ... but they are as they are. We probably can't change the behaviour without breaking stuff. 

Especially have a look at the negative suffixes `:-x` they produce some confusing results. Most of them can be explained, but some are "just strange". ... 

Negative suffixes are probably not used nor documented, but they are possible, so they should be tested.

### CODE REVIEW

**Comment by pmario** [comment]
@matabele ... Can you have a closer look at the tests. ... I know some of them are not intended usecases, but they should be tested anyway.

**Comment by Jermolene** [comment]
Hi @pmario sorry to remind you of #2744, and the passing of @matabele.

**Comment by pmario** [comment]
Ah, So the tests as they are describe the actual behaviour. 

If you have a closer look at the comments, you'll see, where the "strange" things happen. ... We can either define the "wanted" behaviour and fix it OR we can "nail it down" to the existing behaviour. 

Strange things are mainly going on with negative suffixes, which seem to be not intended, except for the "move" operator. .... "move" already [got changed](https://github.com/Jermolene/TiddlyWiki5/pull/2658) in a way, that I don't like too much.

**Comment by Jermolene** [comment]
Hi @pmario indeed some of that behaviour does seem unexpected! My inclination would be to merge these tests so that the current behaviour doesn't get changed inadvertently, but not to make a systematic attempt to "fix" them. Of course, I'd be happy to consider if there's a particular motivation to fix any particular case.

Also, there's again a bunch of eslint-related changes in this PR. I've no objection to updating that stuff across the board, but we probably shouldn't just so it piecemeal.

**Comment by pmario** [comment]
>Also, there's again a bunch of eslint-related changes in this PR. I've no objection to updating that stuff across the board, but we probably shouldn't just so it piecemeal.

Do you use "eslint-like" addOn for your editor?

I could update all files at once, but that would invalidate all >100 existing PRs, which I don't want to happen.

**Comment by pmario** [comment]
The listops operators exist since end of 2015 and we didn't have "functionality issues". So I think, it's OK to merge the tests. So the existing behaviour is "documented" and doesn't change by accident. 

If someone wants to change the behaviour and tests start to fail, we need to discuss the "new" functionality anyway.

**Comment by pmario** [comment]
@Jermolene ... Is there a problem with the new tests?

File: core/modules/filters.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ function parseFilterOperation(operators,filterString,p) {
 				if(rexMatch) {
 					operator.regexp = new RegExp(rexMatch[1], rexMatch[2]);
 // DEPRECATION WARNING
-console.log("WARNING: Filter",operator.operator,"has a deprecated regexp operand",operator.regexp);
+console.log("\nWARNING: Filter",operator.operator,"has a deprecated regexp operand",operator.regexp);
```
Comment (Jermolene) [inline core/modules/filters.js]: Why is this newline being introduced?

Comment (pmario) [reply core/modules/filters.js]: If I run the tests like so: 

```
PS D:\git\tiddly\tiddlywiki\TiddlyWiki5> node .\tiddlywiki.js .\editions\test\ --build index
```
I get this output, which is much better to read, then with no NL and it doesn't create problems in the browser console. The warning is more visible there too. 

```
Randomized with seed 01155
Started
.....................................................................
WARNING: Filter modifier has a deprecated regexp operand /JoeBloggs/

WARNING: Filter modifier has a deprecated regexp operand /Jo/
......................................................
WARNING: Filter modifier has a deprecated regexp operand /JoeBloggs/

WARNING: Filter modifier has a deprecated regexp operand /Jo/
..........................


149 specs, 0 failures
Finished in 0.822 seconds
Randomized with seed 01155 (jasmine --random=true --seed=01155)
```

Without the new-lines the test-output is "scary". IMO with the NLs it's just a warning, which can be ignored.

Comment (pmario) [reply core/modules/filters.js]: With no new-lines, the first thing I saw, was the red text `/JoeBloggs/` and `/Jo/`. The rest of the text is hard to read. So I thought there is a real problem going on. ... 

The NL may avoid this problem for other users in the future.

Comment (Jermolene) [reply core/modules/filters.js]: Hi @pmario that makes sense, but it's still troubling to be changing something unrelated to the core purpose of this PR. I'd prefer to keep it how it is, or to add an option for disabling the warning message.

**Comment by pmario** [comment]
@Jermolene only 1 files changed now.

**Comment by Jermolene** [comment]
Thanks @pmario

**Pull request merged**